Allow subclassing TextView and overriding certain methods

nighthawk opened the pull request:
This is a great library and works really well, and I enjoyed the challenge of wrapping my head around Tree-sitter!

I'm using Runestone's TextView in a project that allows writing expressions including Xcode-like autocompletion that is anchored to the selected text. That requires the ability to subclass TextView (to intercept certain keystrokes for interacting with the autocompletion list) and to access the underlying `UITextInput` protocol (to get the rectangle of the selection, to put the autocompletion list next to it).

For consistency, I made the various methods that are overridden open or that provide UITextInput-like functionality.

simonbs commented:
@nighthawk I'm fine with opening up the class for subclassing. That makes sense.

I'm a little in doubt which direction it makes sense to take to expose some of the underlying UITextInput operations. This PR both exposes the underlying UITextInput instance through the new `textInput` property and it exposes `deleteBackward()` of the underlying text input through the new `deleteBackward()` function.

Would it make sense to avoid exposing the underlying text input and make "proxy functions" for everything?

nighthawk commented:
> Would it make sense to avoid exposing the underlying text input and make "proxy functions" for everything?

Sure, that makes sense. Added that `textInput` was more of a shortcut for not having all those proxy functions, but it's more consistent to have them. 

I've added several proxy functions where TextInputView will return something non-nil.
